src/test/java/calculator/StringCalculator.java

```diff
@@ -0,0 +1,43 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.thymeleaf.util.StringUtils;
+
+public class StringCalculator {
+
+    public static final String DELIMITER = ",|:";
+    public static final String SEPARATOR = "|";
+    public static final Pattern pattern = Pattern.compile("//(.)\n(.*)");
+    public static final int ZERO = 0;
+    public static final int ONE = 1;
+    public static final int TWO = 2;
+    public static final String NEGATIVE_EXCEPTION_MESSAGE = "음수는 계산할 수 없습니다";
+
+    public int calculate(final String stringNumber) {
+        if (StringUtils.isEmpty(stringNumber)) {
+            return ZERO;
+        }
+        final Matcher matcher = pattern.matcher(stringNumber);
+        if (matcher.find()) {
+            final String customDelimiter = matcher.group(ONE);
+            final String targetNumber = matcher.group(TWO);
+            return calculate(targetNumber, DELIMITER + SEPARATOR + customDelimiter);
+        }
+        return calculate(stringNumber, DELIMITER);
+    }
+
+    private int calculate(final String stringNumber, final String delimiter) {
+        return Arrays.stream(stringNumber.split(delimiter))
+            .mapToInt(Integer::parseInt)
+            .peek(this::negativeCheck)
+            .sum();
+    }
+
+    private void negativeCheck(final int number) {
```

Comment:
전달 받은 문자가 음수인지 판단하는 책임을 **새로운 객체**를 생성하여 위임하는것은 어떨까요  ?

```suggestion
    private void checkNegativeNumber(final int number) {
```
일반적으로 메소드는 동사 또는 동사구로 시작하는것을 권장 드려요. 😄

Reply:
맞습니다. 기본적인 내용인데 놓쳤네요. 수정하겠습니다 :)